client/actions/index.js

```diff
@@ -4,6 +4,19 @@ export const GET_MESSAGES = 'GET_MESSAGES'
 export const RECEIVE_MESSAGES = 'RECEIVE_POSTS'
 export const REQUEST_MESSAGES = 'REQUEST_MESSAGES'
 
+export function requestDeleteMessage (id) {
+  return (dispatch) => {
+    request
+      .delete(`/api/v1/messages/${id}`)
+      .end((err, res) => {
+        if (err) {
+          showError(err.message)
```

Comment:
You should put a `return` at the beginning of this line to precent the `dispatch` call on line 15 from happening.